compiler/p/codegen/ControlFlowEvaluator.cpp

```diff
@@ -1252,6 +1254,12 @@ TR::Register *OMR::Power::TreeEvaluator::compareIntsForEquality(TR::InstOpCode::
       site->setNode(node);
       site->setAconstNode(secondChild);
       }
+   else
+      {
+      TR_ASSERT(!(node->isNopableInlineGuard()),"Should not evaluate class or method pointer constants underneath NOPable guards as NOPable guard registers are runtime assumptions and do not need relocation records");
```

Comment:
As we handle the NOPable node using virtual guard helper, how about,
"Should not evaluate class or method pointer constants underneath NOPable guards as they are runtime assumptions handled by virtualGuardHelper");

?